src/AspNetCoreModuleV2/CommonLib/PipeOutputManager.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "stdafx.h"
+
+PipeOutputManager::PipeOutputManager() :
+    m_dwStdErrReadTotal(0),
+    m_hLogFileHandle(INVALID_HANDLE_VALUE),
+    m_hErrReadPipe(INVALID_HANDLE_VALUE),
+    m_hErrWritePipe(INVALID_HANDLE_VALUE),
+    m_hErrThread(INVALID_HANDLE_VALUE),
+    m_fDisposed(FALSE)
+{
+}
+
+PipeOutputManager::~PipeOutputManager()
+{
+    Dispose();
+}
+
+VOID
+PipeOutputManager::Dispose()
```

Comment:
Remove dispose method, move code to finalizer.

Reply:
When you say finalizer, are you referring to c++ finalizers?

Reply:
Yes